### PR TITLE
twopane-navigation library

### PR DESCRIPTION
**Pull request by keilwastaken**

### CODE REVIEW

File: twopane-navigation/src/utilities/constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const hingeWidth: number = 33.6;
```
Comment (kmelmon) [inline twopane-navigation/src/utilities/constants.ts:1]: We should avoid using hard-coded sizes.  Can  this use the hingeWidth property from dualscreeninfo?  This queries the actual device.

File: twopane-navigation/src/components/twoPaneHub/TwoPaneHub.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    utilityStore.isTwoPane(update.isSpanning)
+
+    if (update.isSpanning) {
+      if(paneRects.length < 2)
+      {
+        setPaneRects(update.windowRects)
+      }
+      onePane.mergeToOppositePane();
+    } else {
+    if(paneRects.length < 1) {
```
Comment (kmelmon) [inline twopane-navigation/src/components/twoPaneHub/TwoPaneHub.tsx:56]: else if would be more readable

Comment (kmelmon) [reply twopane-navigation/src/components/twoPaneHub/TwoPaneHub.tsx:56]: oops nevermind

---
In reply to: [476812522](https://github.com/microsoft/react-native-dualscreen/pull/50#discussion_r476812522) [](ancestors = 476812522)

**Comment by kmelmon** [comment]
The readme for the react-native-dualscreen repo should be updated to include mentioning this new control.  See:  [https://github.com/microsoft/react-native-dualscreen/blob/master/README.md](https://github.com/microsoft/react-native-dualscreen/blob/master/README.md)

File: twopane-navigation/src/twoPane/twoPaneStore/tests/twoPane.methods.test.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
+import React from 'react'
+import twoPane from '../twoPane.methods';
+import * as keyActions from '../../../shared/screenStore/keyStore/key.actions';
+import * as headerActions from '../../../shared/screenStore/headerStore/header.actions';
+import * as paneElementActions from '../../../shared/screenStore/paneElementStore/paneElement.action';
```
Comment (kmelmon) [inline twopane-navigation/src/twoPane/twoPaneStore/tests/twoPane.methods.test.tsx:5]: Love to see the tests!  How do you run them?  Also we should setup the CI loop to run them (can be tracked as a separate issue).

Comment (keilwastaken) [reply twopane-navigation/src/twoPane/twoPaneStore/tests/twoPane.methods.test.tsx:5]: sure you just have to run npm run test. to give them a spin currently the code coverage is averaging around 80%. personally i would love to shoot for upper 90's but i figured one thing at a time.

File: twopane-navigation/src/utilities/constants.ts (hunk 1 above)
Comment (keilwastaken) [reply twopane-navigation/src/utilities/constants.ts:1]: very smart, I completely spaced on the dualscreeninfo.hingeWidth property to be honest. adding this now

**Pull request merged**